src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -881,7 +883,27 @@ internal async Task Close(bool connectionClosing, bool async, bool isDisposing)
             case ConnectorState.Executing:
             case ConnectorState.Connecting:
                 if (State != ReaderState.Consumed)
-                    await Consume(async);
+                {
+                    try
+                    {
+                        await Consume(async);
+                    }
+                    catch (Exception ex) when (
+                        ex is OperationCanceledException ||
+                        ex is NpgsqlException && ex.InnerException is TimeoutException)
+                    {
+                        // nothing to do here
+                    }
+                    catch (PostgresException ex)
+                    {
+                        consumeException = ex;
```

Comment:
Tiny tweak: in case there are multiple PostgresExceptions, I suggest capturing the first rather than the last (though this is probably a theoretical scenario only). Also added a comment to explain a bit.

```suggestion
                        // In the case of a PostgresException, the connection is fine and we want to continue consuming the resultset.
                        // Defer throwing the exception until Cleanup is complete below.
                        consumeException ??= ex;
```

Reply:
But there wouldn't be, as we call `consume` only once.

Reply:
Looks even better!

Reply:
Misread the code - you're right.